index.js

```diff
@@ -48,15 +48,32 @@ var pify = module.exports = function (obj, P, opts) {
 		return opts.include ? opts.include.some(match) : !opts.exclude.some(match);
 	};
 
-	var ret = typeof obj === 'function' ? function () {
-		if (opts.excludeMain) {
-			return obj.apply(this, arguments);
-		}
+	var ret;
+	if (typeof obj === 'function') {
+		ret = function () {
+			if (opts.excludeMain) {
+				return obj.apply(this, arguments);
+			}
+
+			return processFn(obj, P, opts).apply(this, arguments);
+		};
+	} else if (opts.inherited) {
+		ret = Object.create(Object.getPrototypeOf(obj));
+	} else {
+		ret = {};
+	}
 
-		return processFn(obj, P, opts).apply(this, arguments);
-	} : {};
+	var props;
+	if (opts.inherited) {
+		props = [];
+		for (var prop in obj) {	// eslint-disable-line guard-for-in
+			props.push(prop);
+		}
+	} else {
+		props = Object.keys(obj);
+	}
```

Comment:
Not sure about this block. If `inherited` is set to `true`, it iterates twice over the properties. Once in the `for...in`, and once in the `reduce` function below.

You could extract the function and do it like this

```js
if(opts.inherited) {
     for (var prop in obj) {
         handler(ret, prop);
     }

     return ret;
}

return Object.keys(obj).reduce(handler, ret);
```

> NOTE: Just a quick sample, might contain issues :)

But before changing this, we should discuss the `inherited` property first.